src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -328,20 +328,22 @@ internal void RemoveFunction(TexlFunction function)
             _functions.RemoveAll(function);
         }
 
-        internal void AddFunctions(TexlFunctionSet functions)
+        internal void AddFunctions(TexlFunctionSet functions, bool makeCopy = false)
         {
             using var guard = _guard.Enter(); // Region is single threaded.
             Inc();
 
-            if (functions._count == 0)
+            if (functions.Count() == 0)
             {
                 return;
             }
-
-            _functions.Add(functions);
+
+            TexlFunctionSet funcsToAdd = makeCopy ? functions.Clone() : functions;

```

Comment:
>         TexlFunctionSet funcsToAdd = makeCopy ? functions.Clone() : functions; [](http://example.com/codeflow?start=3&length=79)

fwiw - I notice that if I switch this to always clone() (and ignore makeCopy), we still get same perf numbers in our benchmark.  

Is that expected?

Reply:
very possible as this method isn't used a lot IIRC